File: 283_Move_Zeroes.java
// O(1) space || O(n) time 
class Solution {
    public void moveZeroes(int[] nums) {
        int nonZeroIndex = 0;
        for(int i = 0; i < nums.length; i++)
        {
            if(nums[i] != 0)
            {
                nums[nonZeroIndex] = nums[i];
                nonZeroIndex++; 
            }
        }
        for(int j = nonZeroIndex; j < nums.length; j++)
        {
            nums[j] = 0; 
        }
    }
}

File: 283_Move_Zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int n = 0; 
        for(int i = 0; i < nums.size(); i++)
        {
            if(nums[i] != 0)
            {
                nums[n] = nums[i];
                n++; 
            }
        }
        for(int j = n; j < nums.size(); j++){
            nums[j] = 0; 
        }
    }
};